compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -981,6 +992,20 @@ fn link_natively(
 
     match prog {
         Ok(prog) => {
+            let is_msvc_link_exe = if let Some(code) = prog.status.code() {
+                sess.target.is_like_msvc
+                && flavor == LinkerFlavor::Msvc(Lld::No)
+                // Respect the command line override
+                && sess.opts.cg.linker.is_none()
+                // Match exactly "link.exe"
+                && linker_path.to_str() == Some("link.exe")
+                // All Microsoft `link.exe` linking error codes are
+                // four digit numbers in the range 1000 to 9999 inclusive
+                && (code < 1000 || code > 9999)
```

Comment:
This was quite intentionally designed to only consider the error cases to decide whether to display the "maybe install VS" message. But now that you're using it in the success case too, this condition will never be true.

Reply:
oh lol oops

good catch ty

Reply:
i've fixed this. i also removed the `cg_options.linker.is_none()` condition, because sometimes people explicitly pass `-C linker=link.exe`. i don't think this should introduce false positives for the Visual Studio check because it's still looking at exit codes and, well, it's still the same file path.